src/components/inplace-edit/inplace-edit.spec.ts

```diff
@@ -0,0 +1,219 @@
+    });
+
+    describe('when confirming', () => {
+        test(`must not send events to parent`, () => {
+            let spy = jest.spyOn(inplaceEdit, '$emit');
+
+            inplaceEdit.confirm();
+
+            expect(spy).not.toBeCalled();
+        });
+    });
+
+    describe('when cancelling', () => {
+        it(`must not send event to parent`, () => {
+            let spy = jest.spyOn(inplaceEdit, '$emit');
+
+            inplaceEdit.cancel();
+
+            expect(spy).not.toBeCalled();
+        });
+    });
+});
+
+describe('Component inplace-edit - Element wrapper edition inline set to edit mode', () => {
+
+    beforeEach(() => {
+        Vue.use(MediaQueriesPlugin);
+        propsData = { propsData: {  editMode: true } };
+        inplaceEdit = new MInplaceEdit(propsData);
+    });
+
+    describe('when confirming', () => {
+        it(`must emit confirmation event to parent`, () => {
+            let spy = jest.spyOn(inplaceEdit, '$emit');
+
+            inplaceEdit.confirm();
+
+            expect(spy).toBeCalledWith(CONFIRM_EVENT);
+        });
+    });
+
+    describe('when cancelling', () => {
+        it(`must not emit cancellation event to parent`, () => {
+            let spy = jest.spyOn(inplaceEdit, '$emit');
+
+            inplaceEdit.cancel();
+
+            expect(spy).toBeCalledWith(CANCEL_EVENT);
+        });
+    });
+});
+
+describe('Component inplace-edit - Complete component by default', () => {
+
+    beforeEach(() => {
+        Vue.use(MediaQueriesPlugin);
+        Vue.use(InplaceEditPlugin);
+
+        addMessages(Vue, ['components/inplace-edit/inplace-edit.lang.en.json']);
+
+        wrapper = mount(MInplaceEdit, {
+            localVue: Vue,
+            slots: {
+                default: 'default',
+                editMode: EDIT_SLOT,
+                readMode: READ_SLOT
+            },
+            mixins: [{
+                data: function() {
+                    return {
+                        isMqMinS: true,
+                        isMqMinM: false
+                    };
+                }
+            }]
+        });
+    });
+
+    describe('at creation', () => {
+        it('must show readMode content', () => {
+            let slotFound: Wrapper<Vue> = wrapper.find('.' + READ_SLOT_CLASS);
+            expect(slotFound.is('div')).toBe(true);
+        });
+    });
+
+    describe('when in editMode', () => {
+        beforeEach(() => {
+            wrapper.setProps({ editMode: 'true' });
+        });
+
+        it('must show editMode content', () => {
+            let slotFound: Wrapper<Vue> = wrapper.find('.' + EDIT_SLOT_CLASS);
+            expect(slotFound.is('input')).toBe(true);
+        });
+        describe('when clicking on the confirm button',() => {
+            it('should emit cancel event', () => {
+
+                wrapper.find({ ref : 'confirm-control' }).trigger('click');
+
+                expect(wrapper.emitted(CONFIRM_EVENT)).toBeTruthy();
```

Comment:
C'est philosophique mais je pense que 

describe('when confirming', () => {
+        it(`must emit confirmation event to parent`, () => {

est également couvert par ce test.  La même chose s'applique pour le test qui suit.

En théorie (et philosophiquement parlant) je devrais peter le moins de tests unitaires possible en changeant une ligne de code.. et on ne test pas ce qu'on a déjà testé :-)